### ISSUE TITLE
Is this broken now?

### ISSUE THREAD

**Issue by sujeetpillai** [kiliman/tailwindui-crawler#83]
I'm unable to use it to download. 

First I got the "Not logged in error" when downloading a section. Since it was looking for my email address in the HTML, the email address is not present as is on that page html anymore. It's in a modal dialogue.

When I bypassed that, I got an error where "component.snippets" was null. The data-page attribute has a javascript object there doesn't have those fields anymore. In fact the iframes for each component now has srcdoc.

**Comment by tecio-first** [comment]
Interesting what exactly is not working ? I used this repo for the first time yesterday and it works locally and as a download action.

**Comment by muesli-dev** [comment]
ref: https://github.com/kiliman/tailwindui-crawler/issues/84
@sujeetpillai

**Comment by toshsan** [comment]
I'm also getting this.

🔐  Logging into tailwindui.com...
🔍  Fetching https://tailwindui.com/login
⏱   247ms (200)
✅  Success!

🗂   Output is ./output
🔍  Fetching https://tailwindui.com/components
⏱   190ms (200)
⏳  Processing /components/marketing/sections/heroes...
🔍  Fetching https://tailwindui.com/components/marketing/sections/heroes
⏱   157ms (200)
**🚫   Not logged in**

**Comment by toshsan** [comment]
Not logged in error after logging in.
> tailwindui-crawler@4.1.0 start
> node index.mjs

🔐  Logging into tailwindui.com...
🔍  Fetching https://tailwindui.com/login
⏱   235ms (200)
✅  Success!

🗂   Output is ./output
🔍  Fetching https://tailwindui.com/components
⏱   182ms (200)
⏳  Processing /components/marketing/sections/heroes...
🔍  Fetching https://tailwindui.com/components/marketing/sections/heroes
⏱   178ms (200)
**🚫   Not logged in**

**Comment by kiliman** [comment]
I haven't checked recently. Let me see.

**Comment by kiliman** [comment]
<img width="638" alt="image" src="https://github.com/kiliman/tailwindui-crawler/assets/47168/1bc75825-ca7c-411e-bb81-aef5eb4fe730">

**Comment by toshsan** [comment]
Hmm. I have tried again on my repo. Same error as i reported above. Could you please share your commit hash to make sure we are on the same codebase?

**Comment by kiliman** [comment]
It's the latest master d3cc6ee9cf2c87145aa2943ff9c7806e4b80855c

Do you have the all-access subscription? That's what I have. I wonder if that's the difference.

**Comment by abdussamadbello** [comment]
@kiliman The issue I noticed are that tailwind is now blocking non-origin access from its CDN (https://tailwindui.com/build/assets/). So we need to download the sources from the headers and iframe and serve them locally.

**Comment by kiliman** [comment]
Yes, I noticed that too. I have a version running locally that I'm still testing. I'll try to get a preview version that you can try out by tomorrow.

**Comment by abdussamadbello** [comment]
Thanks @kiliman ... Youre awesome!

**Comment by abdussamadbello** [comment]
> Yes, I noticed that too. I have a version running locally that I'm still testing. I'll try to get a preview version that you can try out by tomorrow.

Eagerly waiting for updates

**Comment by VIEWVIEWVIEW** [comment]
> Yes, I noticed that too. I have a version running locally that I'm still testing. I'll try to get a preview version that you can try out by tomorrow.

Would be great if you could release the preview version. I copied almost all components by hand in the meantime to have them available when I'm on a plane. However, they updated many, many components to headlessui 2.1, and I definitely don't want to copy all of them by hand again xD

**Comment by jlarmstrongiv** [comment]
Ahh, I was trying to use this package with the new components Adam talked about in the TailwindUI newsletter today.  

Does anyone have any patches, workarounds, or forks?

**Comment by kiliman** [comment]
@jlarmstrongiv Are you asking about the preview mode? Or are you unable to download the components at all?

**Comment by jlarmstrongiv** [comment]
The latest release 4.1.1 is now downloading components and templates 🎉 that wasn’t working for me yesterday

**Comment by kiliman** [comment]
Yeah, they changed the URLS to include the full domain, which broke how the files were stored.

As for the preview, they also changed how it works now. When you change language in the dropdown, it sends a PUT request to their website and then returns a JSON payload with the data. Perhaps they're trying to crack down on crawlers.

I'm still looking at options to support local previews.

**Comment by jlarmstrongiv** [comment]
One thing I also noticed is that the file names may have unexpectedly appended https/

![image](https://github.com/user-attachments/assets/79004cfc-f365-4e59-87ac-bd0bb5c1c54b)
